examples/file-config-solver/file-config-solver.cpp

```diff
@@ -0,0 +1,150 @@
+             }},
+            {"dpcpp",
+             [] {
+                 return gko::DpcppExecutor::create(
+                     0, gko::ReferenceExecutor::create());
+             }},
+            {"reference", [] { return gko::ReferenceExecutor::create(); }}};
+
+    // executor where Ginkgo will perform the computation
+    const auto exec = exec_map.at(executor_string)();  // throws if not valid
+
+    // Read data
+    auto A = share(gko::read<mtx>(std::ifstream("data/A.mtx"), exec));
+    // Create RHS as 1 and initial guess as 0
+    gko::size_type size = A->get_size()[0];
+    auto host_x = vec::create(exec->get_master(), gko::dim<2>(size, 1));
+    auto host_b = vec::create(exec->get_master(), gko::dim<2>(size, 1));
+    for (auto i = 0; i < size; i++) {
+        host_x->at(i, 0) = 0.;
+        host_b->at(i, 0) = 1.;
+    }
+    auto x = vec::create(exec);
+    auto b = vec::create(exec);
+    x->copy_from(host_x);
+    b->copy_from(host_b);
+
+    // Calculate initial residual by overwriting b
+    auto one = gko::initialize<vec>({1.0}, exec);
+    auto neg_one = gko::initialize<vec>({-1.0}, exec);
+    auto initres = gko::initialize<vec>({0.0}, exec);
+    A->apply(one, x, neg_one, b);
+    b->compute_norm2(initres);
+
+    // Copy b again
+    b->copy_from(host_b);
+
+    // Read the json file into ginkgo structure
+    // cg.json: simple-solver
+    // blockjacobi-cg.json: preconditioned-solver
+    // ir.json: iterative-refinement
+    // parilu.json: ilu-preconditioned-solver (by using factoization parameter
+    //              directly)
+    // pgm-multigrid-cg.json: multigrid-preconditioned-solver (set
+    //                        min_coarse_rows additionally due to this small
+    //                        example matrix)
+    // mixed-pgm-multigrid-cg.json: mixed-multigrid-preconditioned-solver
+    //                              (assuming there are always more than one
+    //                              level)
+    auto config = gko::ext::config::parse_json_file(configfile);
+    // Create the registry, which allows passing the existing data into config
+    // This example does not use existing data.
+    auto reg = gko::config::registry();
+    // Create the default type descriptor, which gives the default common type
+    // (value/index) for solver generation. If the solver does not specify value
+    // type, the solver will use these types.
+    auto td = gko::config::make_type_descriptor<ValueType, IndexType>();
+    // generate the linopfactory on the given executors
+    auto solver_gen = gko::config::parse(config, reg, td).on(exec);
+
+    // Create solver
+    const auto gen_tic = std::chrono::steady_clock::now();
+    auto solver = solver_gen->generate(A);
+    exec->synchronize();
+    const auto gen_toc = std::chrono::steady_clock::now();
+    const auto gen_time =
+        std::chrono::duration_cast<std::chrono::nanoseconds>(gen_toc - gen_tic);
```

Comment:
Maybe use ms to avoid conversion later.
```suggestion
        std::chrono::duration_cast<std::chrono::milliseconds>(gen_toc - gen_tic);
```